### DISCUSSION TITLE
Problem setting containerUser

### DISCUSSION THREAD

**Discussion by cbookg** [devcontainers/community#152] [Q&A]
Hi, I'm trying to use some linuxserver images (lscr.io/linuxserver/code-server:latest) as devcontainers.  These images are configured to use user 'abc' as the application user, and /config as the user's home dir.

When I don't set the 'containerUser' property in my devcontainers spec, the /config directory is properly owned by the 'abc' user, but when I specify the 'containerUser' as 'abc' the /config dir ownership is changed to root and the user cannot create/modify any files.

Is this the right user of the containerUser property?  Any idea what would be changing the permission, or creating the home directory with the root user?

I am trying to set this property such that I can install some features (Java and some custom build ones) that will be aware of the application user.

**Comment by bamurtaugh** [comment]
Thanks for opening. The [spec](https://containers.dev/implementors/json_reference/) defines `containerUser` as: 

> Overrides the user for all operations run as inside the container. Defaults to either `root` or the last `USER` instruction in the related Dockerfile used to create the image. If you want any connected tools or related processes to use a different user than the one for the container, see `remoteUser`.

And as specified in https://containers.dev/implementors/json_reference/#remoteUser:

> A dev container configuration will inherit the remoteUser property from the base image it uses.
> 
> Using the [images](https://github.com/devcontainers/images) and [Templates](https://github.com/devcontainers/templates) part of the spec as an example: remoteUser in these images is set to a custom value - you may view an example in the [C++ image](https://github.com/devcontainers/images/blob/main/src/cpp/.devcontainer/devcontainer.json#L26). The [C++ Template](https://github.com/devcontainers/templates/tree/main/src/cpp) will then inherit the custom remoteUser value from [its base C++ image](https://github.com/devcontainers/templates/blob/main/src/cpp/.devcontainer/Dockerfile#L1).

I think setting `"remoteUser": "abc"` would be the right approach here, but @samruddhikhandale to double check!
